### DISCUSSION TITLE
Where can I find the self test scripts for crazyflie?

### DISCUSSION THREAD

**Discussion by kam56** [bitcraze/discussions#646] [Q&A] [answered]
I am doing research about testing emulators and hardware. To do so we need to write some new test cases. Are the self-tests available? I failed to find them.

**Comment by cc299792458** [comment, marked as answer]
Hi bro!
I think what you find is in system.c or stabilizer.c in the crazyflie firmware, maybe you can check it out?

**Reply by kam56** [reply]
Hi, 
Thank you. I found them in system.c :)

**Reply by cc299792458** [reply]
You are welcome. :)
